test/cql-pytest/util.py

```diff
@@ -294,7 +295,10 @@ def __init__(self, cql, key, value):
         self._original_value = None
 
     def __enter__(self):
-        self._original_value = self._cql.execute(f"SELECT value FROM system.config WHERE name='{self._key}'").one().value
+        self._original_value = literal_eval(
```

Comment:
So I left an answer here - https://github.com/scylladb/scylladb/pull/15399#discussion_r1342182141
Without going into too many details, these quirky python calls, like `literal_eval` and `isalnum` are due to the fact how CQL handles lists (replication strategy warn/fail stores lists), and are here just to facilitate conversion from CQL-lists to python-lists and vice-versa (for restoring previous config values).
BTW. I only added support for lists here, not e.g. for UDT etc.

Reply:
Basically, without this modification, we were reading a list as:
```
["SimpleStrategy"]
```
(note the double quotes)
and it was impossible to set the value back, because we were getting:
```
ERROR 2023-10-17 18:31:06,884 [shard 0:stat] storage_proxy - exception during mutation write to 127.9.121.40: boost::program_options::invalid_option_value (the argument ('"SimpleStrategy"') for option is invalid)
```
With this modification we recognize read value as a proper python type (by a call to `literal_eval) and act accordingly:
1. if it's a simple type, i.e. a number, nothing changes,
2. if it's a list, we save the value a bit differently, i.e. just convert it into a comma-separated list (without brackets), and everything works when restoring such value, an example:
```
UPDATE system.config SET value='SimpleStrategy, LocalStrategy, NetworkTopologyStrategy, EverywhereStrategy' WHERE name='replication_strategy_fail_list'
```

Reply:
> I think that this change should be in a separate patch, explaining what it does.

I am not sure about the policy here (but I guess you know better, since you are the policy-man - not to be confused with a 'policeman'), but this is only 3 lines of code, so is it worth a separate PR? Perhaps an extra explanatory comment would be sufficient enough?

Reply:
@nyh so what do you think about it?

Reply:
@nyh ping

Reply:
OK, I think we can leave it as is.